templates/CRM/Core/BillingBlock.js

```diff
@@ -10,13 +10,13 @@
   function civicrm_billingblock_creditcard_helper() {
     $(function() {
       $.each(CRM.config.creditCardTypes, function(key, val) {
-        var html = '<a href="#" title="' + val + '" class="crm-credit_card_type-icon-' + key + '"><span>' + val + '</span></a>';
+        var html = '<a href="#" data-card_type=" + key + " title="' + val + '" class="crm-credit_card_type-icon-' + val['css_key'] + '"><span>' + val['label'] + '</span></a>';
```

Comment:
actually we don't need the data attribute in the end but TBH it would be better next round to use it more & the calculated css class less